pkg/builder/controller.go

```diff
@@ -197,18 +196,16 @@ func (blder *Builder) Build(r reconcile.Reconciler) (controller.Controller, erro
 	return blder.ctrl, nil
 }
 
-func (blder *Builder) project(obj client.Object, proj objectProjection) (client.Object, error) {
+func (blder *Builder) project(obj client.Object, proj objectProjection) (source.Source, error) {
+	src := source.Kind(blder.mgr.GetCache(), obj)
 	switch proj {
 	case projectAsNormal:
-		return obj, nil
+		return src, nil
 	case projectAsMetadata:
-		metaObj := &metav1.PartialObjectMetadata{}
-		gvk, err := getGvk(obj, blder.mgr.GetScheme())
-		if err != nil {
-			return nil, fmt.Errorf("unable to determine GVK of %T for a metadata-only watch: %w", obj, err)
+		if err := source.KindAsPartialMetadata(src, blder.mgr.GetScheme()); err != nil {
```

Comment:
What I meant with the other comment is that it would be nicer to have a `source.NewMetadataKind(obj client.Object, scheme *runtime.scheme)`, right now it requires to first build a `kind` source and then convert it, which is a bit awkward IMHO

Reply:
Agreed, we need more thinking on how we handle the Metadata projection stuff, reverted some changes and instead moved the Kind object in an internal package